jcl/src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -151,7 +151,7 @@ public boolean enqueue() {
  *			or null if object has been cleared.
  */	
 public T get() {
-	if (VM.J9_GC_POLICY != VM.J9_GC_POLICY_METRONOME) {
+	if ((VM.J9_GC_WRITE_BARRIER_TYPE != VM.J9_GC_WRITE_BARRIER_TYPE_SATB) && (VM.J9_GC_WRITE_BARRIER_TYPE != VM.J9_GC_WRITE_BARRIER_TYPE_SATB_AND_OLDCHECK)) {
```

Comment:
We have the option of removing this check entirely given these changes. `getImpl()` with determine what to do and simply return the referent for non SATB configurations

Reply:
I'd remove it, and consider restoring it only if we see perf impact.